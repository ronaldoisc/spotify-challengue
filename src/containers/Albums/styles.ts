import styled from 'styled-components';


export const Wrapper = styled.section`

   display: flex;
   flex-direction: row;
   justify-content:space-between;
   flex-wrap:wrap;
   
`;



export const CardContent=styled.div`
text-align:center;
& h1{
  font-size:1rem;
}
`;
export const Image = styled.img`
width:100%;
object-fit:cover
`;
