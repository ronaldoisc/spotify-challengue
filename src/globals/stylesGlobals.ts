import { createGlobalStyle } from 'styled-components';


export const size = {
    mobileS: '320px',
    mobileM: '375px',
    mobileL: '425px',
    tablet: '768px',
    laptop: '1024px',
    laptopL: '1440px',
    desktop: '2560px'
}
export const device = {
    mobileS: `(min-width: ${size.mobileS})`,
    mobileM: `(min-width: ${size.mobileM})`,
    mobileL: `(min-width: ${size.mobileL})`,
    tablet: `(min-width: ${size.tablet})`,
    laptop: `(min-width: ${size.laptop})`,
    laptopL: `(min-width: ${size.laptopL})`,
    desktop: `(min-width: ${size.desktop})`,
    desktopL: `(min-width: ${size.desktop})`
};

export const GlobalCSS = createGlobalStyle`
  body {
    margin: 0;
    padding: 0;
    background-color: rgb(18,18,18);
    color:white;
    font-family: 'Poppins', sans-serif;
  }
`;
