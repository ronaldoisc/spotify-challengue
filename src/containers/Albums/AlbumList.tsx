import { Wrapper, CardContent, Image } from './styles';
import { Outlet, useNavigate } from 'react-router-dom';
import { Card } from '../../globals/stylesGlobals';
export const AlbumList = ({list}) => {
  
  let navigate = useNavigate();
  //  Function to navigate at track page
  const handleClickCard=(albumId)=>navigate(`/albums/${albumId}/tracks`)
   
  return (
    <Wrapper>
      <Outlet/>
    {
      list.map(item => {
        return <Card key={item.id} onClick={()=>handleClickCard(item.id)}>
          
            <Image src={item.image} alt={item.name}  width={250} height={250}/>
            <CardContent>
              <h1>{item.name}</h1>
              <p>{item.release_date.split("-")[0]}</p>
            </CardContent>
        </Card>
      })
    }
  </Wrapper>
  )
}
