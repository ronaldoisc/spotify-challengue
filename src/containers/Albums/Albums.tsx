import React, { useEffect } from 'react'
import { useParams, Outlet } from 'react-router-dom';
import { useDispatch, useSelector, RootStateOrAny } from 'react-redux';
import { startLoadingArtistAlbums } from '../../redux/modules/albums';

import { Artist } from '../Artist/Artist';
import { AlbumList } from './AlbumList';
import { Loader } from '../../components/Loader';
import { Navbar } from '../../components/Navbar';

export const Albums = () => {
  let params = useParams();
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(startLoadingArtistAlbums(params.artistId))

  }, [dispatch, params])

  const { ui, albums } = useSelector((state: RootStateOrAny) => state);
  const { list } = albums;
  const { isLoading } = ui;

  return (
    <>
      {
        isLoading ?
          <Loader loading={isLoading}/>
          :
          <>
          <Outlet />
            <Navbar title="spotify" />
            <Artist artistId={params.artistId} />
            <h1 className='animate__animated animate__fadeInDown'>Albums</h1>
            <AlbumList list={list} />
          </>
      }

    </>


  )
}
