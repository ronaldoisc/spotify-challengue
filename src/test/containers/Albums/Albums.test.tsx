import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { ThemeProvider } from "styled-components";
import { theme } from '../../../globals/stylesGlobals';
import { Albums } from '../../../containers/Albums/Albums';
import {
    BrowserRouter,

} from "react-router-dom";


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
    albums: {
        list: []
    },
    artist: {
        activeArtist: {
            id: '',
            name: ''
        }
    },
    ui: {
        isLoading: false
    }
};
let store = mockStore(initialState);

describe('test in the albums component', () => {
    test('should show the albums component correctly', async () => {
        const wrapper = mount(
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <BrowserRouter>
                        <Albums />
                    </BrowserRouter>
                </ThemeProvider>

            </Provider>)

        expect(wrapper).toMatchSnapshot();
    })

});