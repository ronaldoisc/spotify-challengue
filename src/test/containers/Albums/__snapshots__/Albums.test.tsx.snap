// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in the albums component should show the albums component 1`] = `
Array [
  <div>
    <header
      className="sc-furwcr iiWkgd"
    >
      <img
        alt="logo"
        height={40}
        src="banner-ico.png"
        width={40}
      />
      <h1
        className="sc-pVTFL VYNMN"
      >
        spotify
      </h1>
    </header>
  </div>,
  <div
    className="sc-gsDKAQ dLKkcs"
  >
    <img
      alt="artist"
      className="sc-dkPtRN vPKLc"
      height={250}
      width={250}
    />
    <div
      className="sc-hKwDye foCxGe"
    >
      <h1 />
      <div
        className="sc-eCImPb lktEJb"
      >
        <p>
          followers: 
           
          <i
            className="material-icons md-light"
          >
            groups 
          </i>
             
        </p>
        <p>
          popularity: 
           
          <i
            className="material-icons md-light"
          >
             star_rate 
          </i>
           
        </p>
      </div>
    </div>
  </div>,
  <h1
    className="animate__animated animate__fadeInDown"
  >
    Albums
  </h1>,
  <section
    className="sc-jRQBWg hFOFKN"
  />,
]
`;
